### PR TITLE
account for unsupported cloud

### PR DESCRIPTION
**Pull request by jeswanke**
stolostron/backlog#22622

Signed-off-by: John Swanke <jswanke@redhat.com>

### CODE REVIEW

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx [hunk 1]
```diff
@@ -362,8 +377,31 @@ export function InstallSubmarinerForm(props: { availableClusters: Cluster[] }) {
...
+                apiVersion: ManagedClusterAddOnApiVersion,
+                kind: ManagedClusterAddOnKind,
+                metadata: {
+                    name: 'submariner',
+                    namespace: 'roks-cluster',
+                },
+                spec: {
+                    installNamespace: 'submariner-operator',
+                },
+            })
```
Comment (dberg1) [inline frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:391]: @jeswanke Why is this required?
The ManagedClusterAddOnKind objects have already been created on line 332 for all unsupported clusters.
I don't think we should have this additional entry created in roks-cluster namespace.
In https://github.com/stolostron/backlog/issues/22622, roks-cluster is just the name (example) of one of the cluster  we are trying to install the submariner addons on.

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx [hunk 2]
```diff
@@ -362,8 +377,31 @@ export function InstallSubmarinerForm(props: { availableClusters: Cluster[] }) {
...
+                apiVersion: BrokerApiVersion,
+                kind: BrokerKind,
+                metadata: {
+                    name: 'submariner-broker',
+                    namespace: 'submariner-broker',
+                },
+                spec: {
+                    globalnetEnabled: false,
+                },
+            }
```
Comment (KevinFCormier) [inline frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:402]: This is also a hard-coded entry that probably shouldn't be in the code. `submariner-broker` would be the broker namespace for a ClusterSet named `submariner`.
